flocker/control/test/test_httpapi.py

```diff
@@ -440,6 +442,289 @@ def created(ignored):
         return creating
 
 
+class UpdatePrimaryDatasetTestsMixin(APITestsMixin):
+    """
+    Tests for the dataset modification endpoint at
+    ``/configuration/datasets/<dataset_id>``.
+    """
+    def test_unknown_dataset(self):
+        """
+        NOT_FOUND is returned if the requested dataset_id doesn't exist.
+        The error includes the requested dataset_id.
+        """
+        unknown_dataset_id = unicode(uuid4())
+        updating = self.assertResponseCode(
+            b"POST",
+            b"/configuration/datasets/%s" % (
+                unknown_dataset_id.encode('ascii'),),
+            {u'primary': self.NODE_A},
+            NOT_FOUND)
+        updating.addCallback(readBody)
+        updating.addCallback(loads)
+
+        def got_result(result):
+            expected_description = u'Dataset not found.'
+            description = result.pop(u"description")
+            self.assertEqual(expected_description, description)
+        updating.addCallback(got_result)
+
+        return updating
+
+    def _test_change_primary(self, dataset, deployment, origin, target):
+        """
+        Helper method which pre-populates the persistence_service with the
+        supplied ``dataset``, makes an API call to move the supplied
+        ``dataset`` from ``origin`` to ``target`` and finally asserts that the
+        API call returned the expected result and that the persistence_service
+        has been updated.
+
+        :param Dataset dataset: The dataset which will be moved.
+        :param Deployment deployment: The deployment that contains the dataset.
+        :param bytes origin: The node IP address of the node that holds the
+            current primary manifestation of the ``dataset``.
+        :param bytes target: The node IP address of the node to which the
+            dataset will be moved.
+        :returns: A ``Deferred`` which fires when all assertions have been
+            executed.
+        """
+        saving = self.persistence_service.save(deployment)
+
+        def saved(ignored):
+            expected_dataset_id = dataset.dataset_id
+            creating = self.assertResponseCode(
```

Comment:
Can we add a utility method that combines `assertResponseCode` with `readBody` and `loads`?